### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/LearnOpenGL/actions/runs/3982020810#summary-10810077863

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/LearnOpenGL/actions/runs/3982020820#summary-10810077895

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

### CODE REVIEW

File: includes/assimp/DefaultLogger.hpp [hunk 1]
```diff
@@ -125,8 +125,8 @@ class ASSIMP_API DefaultLogger :
         unsigned int severity);
 
     // ----------------------------------------------------------------------
-    /** @copydoc Logger::detatchStream */
-    bool detatchStream(LogStream *pStream,
+    /** @copydoc Logger::detachStream */
+    bool detachStream(LogStream *pStream,
```
Comment (jsoref) [inline includes/assimp/DefaultLogger.hpp:129]: This is probably an api break...

File: includes/assimp/Importer.hpp [hunk 2]
```diff
@@ -107,7 +107,7 @@ namespace Assimp    {
 * If you need the Importer to do custom file handling to access the files,
 * implement IOSystem and IOStream and supply an instance of your custom
 * IOSystem implementation by calling SetIOHandler() before calling ReadFile().
-* If you do not assign a custion IO handler, a default handler using the
+* If you do not assign a custom IO handler, a default handler using the
```
Comment (jsoref) [inline includes/assimp/Importer.hpp:110]: not entirely certain...

File: includes/glm/gtx/extended_min_max.hpp [hunk 3]
```diff
@@ -171,7 +171,7 @@ namespace glm
 	/// @tparam T Floating-point scalar types
 	/// @tparam Q Value from qualifier enum
 	///
-	/// @see gtx_extented_min_max
+	/// @see gtx_extended_min_max
 	template<length_t L, typename T, qualifier Q>
 	GLM_FUNC_DECL vec<L, T, Q> fclamp(vec<L, T, Q> const& x, vec<L, T, Q> const& minVal, vec<L, T, Q> const& maxVal);
```
Comment (jsoref) [inline includes/glm/gtx/extended_min_max.hpp:177]: Hanging a comment to note a renamed file below...

Not entirely certain about it...

File: includes/glm/gtx/intersect.inl [hunk 4]
```diff
@@ -117,13 +117,13 @@ namespace glm
 		if (position.y < typename genType::value_type(0) || position.y > typename genType::value_type(1))
 			return false;
 
-		genType Cotengant = cross(Tengant, edge1);
+		genType Cotangent = cross(Tengant, edge1);
```
Comment (jsoref) [inline includes/glm/gtx/intersect.inl:120]: I spent a while puzzling over this one...

File: includes/irrKlang/irrKlang.h [hunk 5]
```diff
@@ -118,7 +118,7 @@
  *	<ul>
  *       <li>ALSA</li>
  *	</ul>
- *  <li>MacOSX (x86 and PPC)</li>
+ *  <li>Mac OS X (x86 and PPC)</li>
```
Comment (jsoref) [inline includes/irrKlang/irrKlang.h:121]: It's possible certain files come from upstreams. I can try to queue sending PRs to them as well...

Note that the modern platform is `macOS` not `Mac OS X`, but it runs on x86_64 and arm...

File: includes/stb_image.h [hunk 6]
```diff
@@ -67,7 +67,7 @@ See final bullet items below for more info on SIMD.
 
 - Added STBI_MALLOC, STBI_REALLOC, and STBI_FREE macros for replacing
 the memory allocator. Unlike other STBI libraries, these macros don't
-support a context parameter, so if you need to pass a context in to
+support a context parameter, so if you need to pass a context into
```
Comment (jsoref) [inline includes/stb_image.h:70]: `in to`/`into` changes have been quite controversial. Happy to drop this or anything else.

File: src/2.lighting/1.colors/1.light_cube.fs [hunk 7]
```diff
@@ -3,5 +3,5 @@ out vec4 FragColor;
 
 void main()
 {
-    FragColor = vec4(1.0); // set alle 4 vector values to 1.0
+    FragColor = vec4(1.0); // set all 4 vector values to 1.0
```
Comment (jsoref) [inline src/2.lighting/1.colors/1.light_cube.fs:6]: Probably?

File: src/2.lighting/1.colors/1.light_cube.fs [hunk 8]
```diff
@@ -3,5 +3,5 @@ out vec4 FragColor;
...
 }
```
Comment (jsoref) [inline src/2.lighting/1.colors/1.light_cube.fs:7]: I personally prefer files to have a newline at eof, but I try not to fix that while making PRs like this one

**Review by jsoref — COMMENTED** [review]
Some of these corrections were automatically suggested by Google Sheets. All fault mine.

**Comment by JoeyDeVries** [comment]
Hey @jsoref looks fine by me, thank you (indeed don't change locations of comments, these are explicit choices).

Did you compile the project before hitting PR. I need to make sure it runs fine before reviewing.

**Comment by jsoref** [comment]
No I didn't compile, sorry. I don't have an open gl environment.

**Comment by JoeyDeVries** [comment]
Alright, let me check in a different branch if this works. Especially as several changes have been made in 3rd party libraries and will have to make sure.

**Pull request merged**